crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -157,26 +162,27 @@ impl RuffSettingsIndex {
 
                 // If the directory is excluded from the workspace, skip it.
                 if let Some(file_name) = directory.file_name() {
-                    if let Some((_, settings)) = index
+                    let settings = index
                         .read()
                         .unwrap()
                         .range(..directory.clone())
                         .rfind(|(path, _)| directory.starts_with(path))
-                    {
-                        if match_exclusion(&directory, file_name, &settings.file_resolver.exclude) {
-                            tracing::debug!("Ignored path via `exclude`: {}", directory.display());
-                            return WalkState::Continue;
-                        } else if match_exclusion(
-                            &directory,
-                            file_name,
-                            &settings.file_resolver.extend_exclude,
-                        ) {
-                            tracing::debug!(
-                                "Ignored path via `extend-exclude`: {}",
-                                directory.display()
-                            );
-                            return WalkState::Continue;
-                        }
+                        .map(|(_, settings)| settings.clone())
```

Comment:
Are these clones necessary?

Reply:
I think these are all `Arc<RuffSettings>` so this should be cheap.

Reply:
Yes, these are all `Arc`s

Reply:
It otherwise throws "rustc: temporary value dropped while borrowed" if I use a reference

```rs
                    let settings = index
                        .read()
                        .unwrap()
                        .range(..directory.clone())
                        .rfind(|(path, _)| directory.starts_with(path))
                        .map(|(_, settings)| settings)
                        .unwrap_or(&fallback);
```